### PR TITLE
Add ability to override vcpkg_ports, add error checking to spack_configs

### PR DESCRIPTION
**Pull request by white238**
Needed to move axom to a submodule.  Currently testing it.

### CODE REVIEW

File: uberenv.py [hunk 1]
```diff
@@ -365,7 +365,29 @@ def setup_paths_and_dirs(self):
 
         UberEnv.setup_paths_and_dirs(self)
 
-        self.ports = pjoin(self.uberenv_path, "vcpkg_ports","*")
+        # Find path to vcpkg ports
```
Comment (kennyweiss) [inline uberenv.py:368]: It seems like we should have a utility function to get the item from the command line with a fallback to uberenv_config

File: uberenv.py [hunk 2]
```diff
@@ -517,8 +538,12 @@ def setup_paths_and_dirs(self):
 
         UberEnv.setup_paths_and_dirs(self)
 
-        # Spack yaml configs path (compilers.yaml, packages.yaml, etc.)
+        # Find Spack yaml configs path (compilers.yaml, packages.yaml, etc.)
+
+        # Next to uberenv.py (backwards compatility)
```
Comment (kennyweiss) [inline uberenv.py:543]: 👍  for the comments

**Review by kennyweiss — APPROVED** [review]

File: uberenv.py [hunk 3]
```diff
@@ -365,7 +365,29 @@ def setup_paths_and_dirs(self):
 
         UberEnv.setup_paths_and_dirs(self)
 
-        self.ports = pjoin(self.uberenv_path, "vcpkg_ports","*")
+        # Find path to vcpkg ports
+        _errmsg = ""
+        if not self.opts["vcpkg_ports_path"] is None:
```
Comment (adrienbernede) [inline uberenv.py:370]: Just below is used the much more human-readable `if foo is not None`. I suggest we use the same syntax here.

File: uberenv.py [hunk 4]
```diff
@@ -530,9 +555,14 @@ def setup_paths_and_dirs(self):
         # Test if the override option was used (--spack-config-dir)
         self.spack_config_dir = self.opts["spack_config_dir"]
         if self.spack_config_dir is None:
+            # If command line option is not used, search for platform under
+            # given directory
             uberenv_plat = self.detect_platform()
             if not uberenv_plat is None:
```
Comment (adrienbernede) [inline uberenv.py:561]: Another candidate for the `is not None` syntax (out of scope, but I can see it... ;) ).

**Review by adrienbernede — APPROVED** [review]
Looks good to me. Only minor comments.

File: uberenv.py (hunk 3 above)
Comment (white238) [reply uberenv.py]: did this and all other instances in script.

File: uberenv.py (hunk 1 above)
Comment (white238) [reply uberenv.py:368]: I agree.  We should do a followup PR to do smaller cleanup things like this as well as add some CI checks to PRs.

**Comment by white238** [comment]
@kennyweiss this seems to be passing the regular tests on axom. Could you test vcpkg since I cant?

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -15,6 +15,7 @@ The Uberenv project release numbers follow [Semantic Versioning](http://semver.o
   inside of your project's source repository.
 - Adds support for Windows builds using [Vcpkg].
 - Adds the `--triplet` command line argument for setting the Vcpkg build configuration.
+- Adds the `--vcpkg-ports-path` command line argument for setting the path to the vcpkg ports directory.
```
Comment (kennyweiss) [inline RELEASE-NOTES.md:18]: I added a missing command line argument to support supplying a path to the vcpkg ports directory

File: uberenv.py [hunk 6]
```diff
@@ -388,7 +416,7 @@ def clone_repo(self):
             clone_opts = ("-c http.sslVerify=false " 
                           if self.opts["ignore_ssl_errors"] else "")
 
-            clone_cmd =  "git {0} clone --single-branch --depth=1 -b {1} {2} vcpkg".format(clone_opts, vcpkg_branch,vcpkg_url)
+            clone_cmd =  "git {0} clone --single-branch -b {1} {2} vcpkg".format(clone_opts, vcpkg_branch,vcpkg_url)
```
Comment (kennyweiss) [inline uberenv.py:419]: Note: `--depth=1` broke our ability to use a custom commit

**Review by kennyweiss — APPROVED** [review]

**Comment by white238** [comment]
This was fully tested in Axom so should be fine to merge now.

File: uberenv.py (hunk 6 above)
Comment (white238) [reply uberenv.py:419]: Git cloning should also be merged into a function between the spack and vcpkg... next PR

File: uberenv.py (hunk 1 above)
Comment (white238) [reply uberenv.py:374]: logged #50 so we don't forget.

File: uberenv.py (hunk 6 above)
Comment (white238) [reply uberenv.py:419]: #49

**Pull request merged**